spec/classes/profile/keepalived_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+require_relative '../../support/contexts/with_mocked_nodes'
+
+describe 'nebula::profile::keepalived' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:my_ip) { Faker::Internet.ip_v4_address }
+      let(:facts) do
+        os_facts.merge(
+          datacenter: 'somedc',
+          networking: {
+            ip: my_ip,
+            primary: 'eth0',
+          },
+          hostname: 'thisnode',
+        )
+      end
+
+      let(:thisnode) { { 'ip' => facts[:networking][:ip], 'hostname' => facts[:hostname] } }
+      let(:scotch) { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'scotch' } }
+      let(:soda)   { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'soda' } }
+      let(:base_file) { '/etc/keepalived/keepalived.conf' }
+      let(:service) { 'keepalived' }
+
+      include_context 'with mocked puppetdb functions', 'somedc', %w[thisnode scotch soda]
+
+      describe 'packages' do
+        it { is_expected.to contain_package('keepalived') }
+        it { is_expected.to contain_package('ipset') }
+      end
+
+      describe 'sysctl conf' do
+        let(:file) { '/etc/sysctl.d/keepalived.conf' }
+
+        it { is_expected.to contain_file(file).with(ensure: 'present') }
+        it { is_expected.to contain_file(file).with(mode: '0644') }
+
+        it 'says it is managed by puppet' do
+          is_expected.to contain_file(file).with_content(
+            %r{\A# Managed by puppet \(nebula\/profile\/keepalived\/sysctl\.conf\.erb\)\n},
+          )
+        end
```

Comment:
I haven't been testing this, but I'm open to the idea. My gut reaction is that this test is pretty brittle, though. I guess I'd like it better if it didn't specifically test the template path.

Reply:
I removed the template path.